client/modules/IDE/pages/IDEView.jsx

```diff
@@ -428,122 +338,7 @@ IDEView.propTypes = {
-    shareModalProjectUsername: PropTypes.string.isRequired,
-    previousPath: PropTypes.string.isRequired,
-    previewIsRefreshing: PropTypes.bool.isRequired,
-    isPlaying: PropTypes.bool.isRequired,
-    isAccessibleOutputPlaying: PropTypes.bool.isRequired,
-    projectOptionsVisible: PropTypes.bool.isRequired,
-    preferencesIsVisible: PropTypes.bool.isRequired,
-    modalIsVisible: PropTypes.bool.isRequired,
-    uploadFileModalVisible: PropTypes.bool.isRequired,
-    newFolderModalVisible: PropTypes.bool.isRequired,
-    justOpenedProject: PropTypes.bool.isRequired,
-    sidebarIsExpanded: PropTypes.bool.isRequired,
-    consoleIsExpanded: PropTypes.bool.isRequired,
-    unsavedChanges: PropTypes.bool.isRequired
-  }).isRequired,
-  stopSketch: PropTypes.func.isRequired,
-  project: PropTypes.shape({
-    id: PropTypes.string,
-    name: PropTypes.string.isRequired,
-    owner: PropTypes.shape({
-      username: PropTypes.string,
-      id: PropTypes.string
-    }),
-    updatedAt: PropTypes.string
-  }).isRequired,
-  preferences: PropTypes.shape({
-    autosave: PropTypes.bool.isRequired,
-    fontSize: PropTypes.number.isRequired,
-    linewrap: PropTypes.bool.isRequired,
-    lineNumbers: PropTypes.bool.isRequired,
-    lintWarning: PropTypes.bool.isRequired,
-    textOutput: PropTypes.bool.isRequired,
-    gridOutput: PropTypes.bool.isRequired,
-    theme: PropTypes.string.isRequired,
-    autorefresh: PropTypes.bool.isRequired,
-    language: PropTypes.string.isRequired,
-    autocloseBracketsQuotes: PropTypes.bool.isRequired,
-    autocompleteHinter: PropTypes.bool.isRequired
-  }).isRequired,
-  closePreferences: PropTypes.func.isRequired,
-  setAllAccessibleOutput: PropTypes.func.isRequired,
-  selectedFile: PropTypes.shape({
-    id: PropTypes.string.isRequired,
-    content: PropTypes.string.isRequired,
-    name: PropTypes.string.isRequired
-  }).isRequired,
-  htmlFile: PropTypes.shape({
-    id: PropTypes.string.isRequired,
-    name: PropTypes.string.isRequired,
-    content: PropTypes.string.isRequired
-  }).isRequired,
-  expandSidebar: PropTypes.func.isRequired,
-  collapseSidebar: PropTypes.func.isRequired,
-  cloneProject: PropTypes.func.isRequired,
-  expandConsole: PropTypes.func.isRequired,
-  collapseConsole: PropTypes.func.isRequired,
-  updateFileContent: PropTypes.func.isRequired,
-  closeNewFolderModal: PropTypes.func.isRequired,
-  closeNewFileModal: PropTypes.func.isRequired,
-  closeShareModal: PropTypes.func.isRequired,
-  closeKeyboardShortcutModal: PropTypes.func.isRequired,
-  autosaveProject: PropTypes.func.isRequired,
-  setPreviousPath: PropTypes.func.isRequired,
-  showErrorModal: PropTypes.func.isRequired,
-  hideErrorModal: PropTypes.func.isRequired,
-  clearPersistedState: PropTypes.func.isRequired,
-  startSketch: PropTypes.func.isRequired,
-  closeUploadFileModal: PropTypes.func.isRequired,
-  t: PropTypes.func.isRequired,
-  isUserOwner: PropTypes.bool.isRequired
+  }).isRequired
 };
```

Comment:
Look at all those props gone!  So satisfying.  You actually don’t need ANY props anymore because you can get the params and location from react-router hooks.

Reply:
Yess 😙